lib/models/command.js

```diff
@@ -288,7 +288,11 @@ let Command = CoreObject.extend({
 
       if (this.works === 'insideProject') {
         if (!this.project.hasDependencies()) {
-          throw new SilentError('node_modules appears empty, you may need to run `npm install`');
+          let installInstuction = '`npm install`';
+          if (this.project.hasYarnLockFile()) {
```

Comment:
@rwjblue, in [this commit](https://github.com/ember-cli/ember-cli/pull/6888/commits/ea2f6dcbd96783d7e5d67cf68de38022a8eafcfc) I will check it directly. To test this behaviour, I am renaming the `ember-cli` project's actual `yarn.lock` file before the test runs and after the test, rewrite it back to the original name.
Could you advise me if this is a good approach, or might it be to risky? If so, what would you suggest for testing?